### PR TITLE
expression: rewrite builtin function: TO_SECONDS

### PR DESCRIPTION
**Pull request by spongedu**
for #4080

### CODE REVIEW

File: expression/builtin_time.go [hunk 1]
```diff
@@ -2842,35 +2842,37 @@ func (c *toSecondsFunctionClass) getFunction(args []Expression, ctx context.Cont
...
 }
 
-// eval evals a builtinToSecondsSig.
+// evalInt evals a builtinToSecondsSig.
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_to-seconds
-func (b *builtinToSecondsSig) eval(row []types.Datum) (d types.Datum, err error) {
-	args, err := b.evalArgs(row)
+func (b *builtinToSecondsSig) evalInt(row []types.Datum) (int64, bool, error) {
+	arg, isNull, err := b.args[0].EvalString(row, b.getCtx().GetSessionVars().StmtCtx)
 	if err != nil {
```
Comment (zz-jason) [inline expression/builtin_time.go]:
```go
if isNull || err != nil {
    return 0, isNull, errors.Trace(err)
}
```

File: expression/builtin_time.go [hunk 2]
```diff
@@ -2842,35 +2842,37 @@ func (c *toSecondsFunctionClass) getFunction(args []Expression, ctx context.Cont
...
+		return 0, true, errors.Trace(err)
 	}
-	if args[0].IsNull() {
-		return
+	if isNull {
+		return 0, true, nil
 	}
-	sc := b.ctx.GetSessionVars().StmtCtx
-	date, err := convertDatumToTime(sc, args[0])
+	date, err := types.ParseTime(arg, mysql.TypeDatetime, types.MaxFsp)
```
Comment (zz-jason) [inline expression/builtin_time.go]: maybe a better way is to `bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpInt, tpTime)` in line 2845 and call `EvalTime` in line 2860

File: expression/builtin_time.go (hunk 1 above)
Comment (spongedu) [inline expression/builtin_time.go]: ok

File: expression/builtin_time.go (hunk 2 above)
Comment (spongedu) [inline expression/builtin_time.go]: ok

**Pull request merged**